src/server/api/routers/groups.ts

```diff
@@ -26,10 +28,21 @@ export const groupRouter = createTRPCRouter({
     };
   }),
 
-  fetchGroups: publicProcedure.query(async ({ ctx }) => {
+  fetchGroups: protectedProcedure.query(async ({ ctx }) => {
+    const router = useRouter();
+    const userId = ctx.session?.user?.id;
+
+    if (!userId) {
+      console.log('no user id')
+      void router.replace("/auth/onboarding")
```

Comment:
I could use some clarification with this redirect

Reply:
In the event there is no user id this would send the user to a new page. I'm not sure this works as intended though. Because this is a protectedProcedure there should always be a user id and always be a session. So this redirect should never actually happen if I'm reading this correctly.

Reply:
True that makes sense, still wrapping my head around protected vs public

Reply:
Protected requires a user to be logged in. Public does not require a user to be logged in. 
If someone that is not logged in tries to use a protected route it just won't work.

Reply:
So it wont even step into the function, correct?

Reply:
It will error out.  You can pass a second parameter object onError, and then set GlobalContext modal to true, which will redirect them to a login screen.